dict_inventory.py

```diff
@@ -0,0 +1,45 @@
+"""
+  Sub class of BaseInventory, implemented with dictionary:
+  key: id of the product
+  value: the Product object
+"""
+
+from base_inventory import BaseInventory
+from product import Product
+
+
+class DictInventory(BaseInventory):
+
+  def __init__(self):
+    BaseInventory.__init__(self)
+    self.products = {}
+    self.test_init()
+
+
+  def get_product(self, product_id):
+    if product_id in self.products:
+      return self.products[product_id]
+    else:
+      return None
+
+
+  def get_all(self):
+    result = {}
+    for k in self.products:
+      result[k] = self.products[k].__str__()
+    return result
+
+
+  def put_product(self, id, info):
+    self.products[id] = Product(id, info['location_id'], info['used'],
+                                info['new'], info['open_box'], info['restock_level'])
+    return True
+
+
+  def delete_product(self, product_id):
+    del self.products[product_id]
```

Comment:
Good catch, Thanks! 
And fixed.
Actually del will cast keyError for product id not found
but I we should explicitly catch the error and print with custom error message.